metaflow/cli.py

```diff
@@ -101,6 +110,82 @@ def logger(body='', system_msg=False, head='', bad=False, timestamp=True):
                 fg=LOGGER_BAD_COLOR if bad else None)
 
 
+def get_ordered_steps(graph, type='linear'):
+    """
+    Returns the ordered step names in the graph (FlowGraph) from start step to end step as a list of strings containing the
+    step names.
+
+    Note: All MF graphs start at the "start" step and end with the "end" step (both of which are mandatory).
+    """
+
+    ordered_steps = ['start']
+    current_node_name = 'start'
+
+    # This is not an ideal way to iterate over the graph, but it's the (easy+)right thing to do for now.
+    # This may need to change as work on improvements.
+    while current_node_name != 'end':
+        for node in graph:
+            if node.name == current_node_name:
+                if current_node_name != 'end':
+                    current_node_name = node.out_funcs[0]
+                    ordered_steps.append(current_node_name)
+                    break
+    return ordered_steps
```

Comment:
I actually just need a static list here and it's okay for it to be stored in memory as well because I'm constructing the entire pipeline spec before it actually runs. My understanding is that I will not be getting any additional value using a yield there.

Reply:
Yeah makes sense, I was thinking ahead to nice way to walk the graph for supporting non-linear operations but happy to defer that to when that is the PR goal 😄